fix flakiness in update_hint_benchmark.py

laithsakka opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #134649
* #134635
* #133834
```
compile time instruction count for iteration 1 is 10732129038
compile time instruction count for iteration 2 is 10719776783
compile time instruction count for iteration 3 is 10729546868
compile time instruction count for iteration 4 is 10737655132
compile time instruction count for iteration 5 is 10732564252
compile time instruction count for iteration 6 is 10728721234
compile time instruction count for iteration 7 is 10733354271
compile time instruction count for iteration 8 is 10719588972
compile time instruction count for iteration 9 is 10706311856
```
1. add torch.manual_seed(0), inputs was not the same across iterations
2. disable gc. 
3. remove loop (not needed since compilation happen once only)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

laithsakka committed:
fix flakiness in update_hint_benchmark.py

[ghstack-poisoned]

laithsakka committed:
Update on "fix flakiness in update_hint_benchmark.py"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "fix flakiness in update_hint_benchmark.py"


1. add torch.manual_seed(0), inputs was not the same across iterations
2. disable gc. 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

oulgen commented on benchmarks/dynamo/pr_time_benchmarks/benchmarks/update_hint_benchmark.py:
```diff
@@ -34,8 +37,7 @@ def f(a, b):
 
-        for i in range(1000):
-            f(self.input, self.splits)
+        f(self.input, self.splits)
```
why are you running it only once now?

laithsakka replied:
in recent changes we change this to measure compile time, since the compilation is cached there is not value nothing will be measured in subsequent runs

aorenste replied:
This may change when #131843 (or something similar) lands enabling some number of warmup runs before we actually compile.

laithsakka replied:
I mean that would require changing alot, I can add a loop the back though if you guys want me to I do not have strong opinion against it.

do you you have more concerns with the diff?

laithsakka replied:
I will add a check that instruction counter is not zero, the reason is that
otherwise we would have to wrap all benchmarks that we ever add with loops.
ideally the diff above should add a flag or something to disable it
because for example all unit tests wont be wrapped in loops 

let me add the check and update the diff

laithsakka committed:
Update on "fix flakiness in update_hint_benchmark.py"

```
compile time instruction count for iteration 1 is 10732129038
compile time instruction count for iteration 2 is 10719776783
compile time instruction count for iteration 3 is 10729546868
compile time instruction count for iteration 4 is 10737655132
compile time instruction count for iteration 5 is 10732564252
compile time instruction count for iteration 6 is 10728721234
compile time instruction count for iteration 7 is 10733354271
compile time instruction count for iteration 8 is 10719588972
compile time instruction count for iteration 9 is 10706311856
```
1. add torch.manual_seed(0), inputs was not the same across iterations
2. disable gc. 
3. remove loop (not needed since compilation happen once only)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka commented:
@pytorchbot  merge

laithsakka commented:
@pytorchbot merge
